python/pyspark/pandas/indexes/multi.py

```diff
@@ -375,6 +375,15 @@ def name(self) -> Name:
     def name(self, name: Name) -> None:
         raise PandasNotImplementedError(class_name="pd.MultiIndex", property_name="name")
 
+    @property
+    def dtypes(self) -> Series:
+        return pd.Series(
+            [field.dtype for field in self._internal.index_fields],
+            index=pd.Index(
+                [name if len(name) > 1 else name[0] for name in self._internal.index_names]
+            ),
+        )
```

Comment:
Can we create pandas-on-Spark Series rather than pandas Series ??

For example,

```python
return ps.Series(
    [field.dtype.name for field in self._internal.index_fields],
    index=[name if len(name) > 1 else name[0] for name in self._internal.index_names])
```

Reply:
Thank you, I will update. I copied behavior of ps.DataFrame.dtypes. Should we change it too?
https://github.com/apache/spark/blob/d6786e036d610476a3be0fca5b16ba819dcbc013/python/pyspark/pandas/frame.py#L6276

Reply:
hmm, if we return ```ps.Series```, we can not use index type hint for now. Can we use ```pd.Series```

``` python
>>> ps.DataFrame[psdf.index.dtypes, psdf.dtypes]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/u02/spark/python/pyspark/pandas/frame.py", line 12097, in __class_getitem__
    return create_tuple_for_frame_type(params)
  File "/u02/spark/python/pyspark/pandas/typedef/typehints.py", line 729, in create_tuple_for_frame_type
    return Tuple[_extract_types(params)]
  File "/u02/spark/python/pyspark/pandas/typedef/typehints.py", line 760, in _extract_types
    index_params = _to_tuple_of_params(index_params)
  File "/u02/spark/python/pyspark/pandas/typedef/typehints.py", line 833, in _to_tuple_of_params
    params = tuple(params)
  File "/u02/spark/python/pyspark/pandas/series.py", line 6366, in __iter__
    return MissingPandasLikeSeries.__iter__(self)
  File "/u02/spark/python/pyspark/pandas/missing/__init__.py", line 23, in unsupported_function
    raise PandasNotImplementedError(
pyspark.pandas.exceptions.PandasNotImplementedError: The method `pd.Series.__iter__()` is not implemented. If you want to collect your data as an NumPy array, use 'to_numpy()' instead.
```

Reply:
I reverted to return ```pd.Series```

Reply:
> hmm, if we return `ps.Series`, we can not use index type hint for now. Can we use `pd.Series` ?
> 
> ```python
> >>> ps.DataFrame[psdf.index.dtypes, psdf.dtypes]
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/u02/spark/python/pyspark/pandas/frame.py", line 12097, in __class_getitem__
>     return create_tuple_for_frame_type(params)
>   File "/u02/spark/python/pyspark/pandas/typedef/typehints.py", line 729, in create_tuple_for_frame_type
>     return Tuple[_extract_types(params)]
>   File "/u02/spark/python/pyspark/pandas/typedef/typehints.py", line 760, in _extract_types
>     index_params = _to_tuple_of_params(index_params)
>   File "/u02/spark/python/pyspark/pandas/typedef/typehints.py", line 833, in _to_tuple_of_params
>     params = tuple(params)
>   File "/u02/spark/python/pyspark/pandas/series.py", line 6366, in __iter__
>     return MissingPandasLikeSeries.__iter__(self)
>   File "/u02/spark/python/pyspark/pandas/missing/__init__.py", line 23, in unsupported_function
>     raise PandasNotImplementedError(
> pyspark.pandas.exceptions.PandasNotImplementedError: The method `pd.Series.__iter__()` is not implemented. If you want to collect your data as an NumPy array, use 'to_numpy()' instead.
> ```

Yeah, I think it's normal because we don't support the iteration for `Series` yet.

I wonder if we can say the length of MultiIndex is always expected to be small.

If the length of MultiIndex is always expected to be small, it's okay to just return the pandas Series.

Otherwise, we should return pandas-on-Spark Series.

WDYT, @HyukjinKwon @ueshin @xinrong-databricks ?

Reply:
Yeah, I think it's fine.